lib/phoenix/controller.ex

```diff
@@ -255,19 +269,28 @@ defmodule Phoenix.Controller do
       iex> layout(conn)
       {AppView, :print}
 
+  Raises `Plug.Conn.AlreadySentError` if the conn was already sent.
   """
   @spec put_layout(Plug.Conn.t, {atom, binary} | binary | false) :: Plug.Conn.t
-  def put_layout(conn, layout)
+  def put_layout(%Plug.Conn{state: state} = conn, layout) do
+    if state in @unsent do
+      _put_layout(conn, layout)
+    else
+      raise Plug.Conn.AlreadySentError
+    end
+  end
```

Comment:
You can do this as:

```elixir
def put_layout(%Plug.Conn{state: state} = conn, layout) when state in @unsent do
  _put_layout(...)
end

def put_layout(_, _) do
  raise Plug.Conn.AlreadySentError
end
```

But in my opinion it does not offer anything compared to your solution.